src/stores/room-list/MessagePreviewStore.ts

```diff
@@ -176,7 +176,8 @@ export class MessagePreviewStore extends AsyncStoreWithClient<IState> {
 
         if (payload.action === 'MatrixActions.Room.timeline' || payload.action === 'MatrixActions.Event.decrypted') {
             const event = payload.event; // TODO: Type out the dispatcher
-            if (!this.previews.has(event.getRoomId()) || !event.isLiveEvent) return; // not important
+            const isHistoricalEvent = payload.hasOwnProperty("isLiveEvent") && !payload.isLiveEvent
```

Comment:
The RoomListStore just uses `!isLiveEvent` rather than checking for the property, any reason why here is different?

Reply:
```
const eventPayload = (<any>payload); // TODO: Type out the dispatcher types
// Ignore non-live events (backfill)
if (!eventPayload.isLiveEvent || !payload.isLiveUnfilteredRoomTimelineEvent) return;
```

The above is what is in `RoomListStore`, the first line seems to reassign that value to a new variable without mutating it to make sure that TypeScript does not complain.
That's probably where I got confused too 

But `isLiveEvent` is not a property member of `MatrixEvent` but part of the action payload defined in `MatrixActionCreators::createRoomTimelineAction`

Reply:
Right, but I'm just saying why not use `!payload.isLiveEvent` rather than bothering with hasOwnProperty?
Is it ever not present where we'd want to assume true? Can't we just use `!== false` there then?

Reply:
@gsouquet I see you re-requested review but this thread is unresolved

Reply:
I did this because `isLiveEvent` only exists for the `MatrixActions.Room.timeline`. However the decrypted action does not have that property member. Because `undefined` is falsy we would ignore all those events